### PR TITLE
22628-merge-latest-FFI-Kernel-changes

### PR DESCRIPTION
**Pull request by estebanlm**
merged FFI-Kernel-eem.56

https://pharo.manuscript.com/f/cases/22628/merge-latest-FFI-Kernel-changes

### CODE REVIEW

**Comment by jecisc** [comment]
We have this in the logs:

```
 Looking for module  ... loaded...computing accessor depths...donewarning, variable translatedMethodCache doesn't exist or has already been removed
[32] warning, variable simulator doesn't exist or has already been removed
[32] warning, variable translatedMethodCache doesn't exist or has already been removed
[32] 'Errors in script loaded from /builds/workspace/and_branch_Pipeline_PR-1952-7YKVG53IZ6XSDLJB7TRX57WGI2TTJQLFXC2BHRYUI77ONEDJWBRA/32/bootstrap/scripts/bootstrap.st'
[32] MessageNotUnderstood: EPUndeclaredBinding>>markWrite
[32] EPUndeclaredBinding(Object)>>doesNotUnderstand: #markWrite
[32] OCASTSemanticAnalyzer>>visitAssignmentNode:
[32] RBAssignmentNode>>acceptVisitor:
[32] OCASTSemanticAnalyzer(RBProgramNodeVisitor)>>visitNode:
[32] [ :each | self visitNode: each ] in OCASTSemanticAnalyzer>>visitSequenceNode: in Block: [ :each | self visitNode: each ]
[32] OrderedCollection>>do:
[32] OCASTSemanticAnalyzer>>visitSequenceNode:
[32] RBSequenceNode>>acceptVisitor:
[32] OCASTSemanticAnalyzer(RBProgramNodeVisitor)>>visitNode:
[32] OCASTSemanticAnalyzer>>visitInlinedBlockNode:
[32] OCASTSemanticAnalyzer>>visitBlockNode:
[32] RBBlockNode>>acceptVisitor:
[32] OCASTSemanticAnalyzer(RBProgramNodeVisitor)>>visitNode:
[32] [ :each | self visitNode: each ] in OCASTSemanticAnalyzer(RBProgramNodeVisitor)>>visitMessageNode: in Block: [ :each | self visitNode: each ]
[32] OrderedCollection>>do:
[32] OCASTSemanticAnalyzer(RBProgramNodeVisitor)>>visitMessageNode:
[32] RBMessageNode>>acceptVisitor:
[32] OCASTSemanticAnalyzer(RBProgramNodeVisitor)>>visitNode:
[32] [ :each | self visitNode: each ] in OCASTSemanticAnalyzer>>visitSequenceNode: in Block: [ :each | self visitNode: each ]
[32] OrderedCollection>>do:
[32] OCASTSemanticAnalyzer>>visitSequenceNode:
[32] RBSequenceNode>>acceptVisitor:
[32] OCASTSemanticAnalyzer(RBProgramNodeVisitor)>>visitNode:
[32] OCASTSemanticAnalyzer>>visitMethodNode:
[32] RBMethodNode>>acceptVisitor:
[32] OCASTSemanticAnalyzer(RBProgramNodeVisitor)>>visitNode:
[32] OCASTSemanticAnalyzer>>analyze:
[32] RBMethodNode>>doSemanticAnalysis
[32] RBMethodNode>>doSemanticAnalysisInContext:
```

**Comment by MarcusDenker** [comment]
what is EPUndeclaredBinding ?

**Comment by estebanlm** [comment]
I have no idea :)

File: src/FFI-Kernel/ExternalStructure.class.st [hunk 1]
```diff
@@ -242,6 +276,17 @@ ExternalStructure class >> fromHandle: aHandle [
 	^self basicNew setHandle: aHandle
 ]
 
+{ #category : #'system startup' }
+ExternalStructure class >> install [
+	"Resuming the image on another architecture may require a re-compilation of structure layout."
+	| newPlatform |
+	newPlatform := Smalltalk platformName.
+	PreviousPlatform = newPlatform
```
Comment (pavel-krivanek) [inline src/FFI-Kernel/ExternalStructure.class.st:284]: I guess that the Undeclared report is related to this PreviousPlatform

**Comment by Ducasse** [comment]
so what do we do with this PR?

**Comment by estebanlm** [comment]
we move it to Pharo8 and we will see there. 
is not urgent anyway.

**Comment by Ducasse** [comment]
Esteban what do we do with this PR? we close it and you reopen it when a new version gets ready?

**Pull request closed without merging**